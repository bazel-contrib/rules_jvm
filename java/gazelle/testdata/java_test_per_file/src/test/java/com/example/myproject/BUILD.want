load("@rules_java//java:defs.bzl", "java_test")

java_test(
    name = "AppTest",
    srcs = ["AppTest.java"],
    _gazelle_imports = [
        "com.example.myproject",
        "java.lang.Exception",
        "org.hamcrest.MatcherAssert", # TODO: per-file tests should have per-test imports, excluding this one
        "org.hamcrest.Matchers", # TODO: per-file tests should have per-test imports, excluding this one
        "org.junit.Assert",
        "org.junit.Test",
    ],
    _java_packages = ["com.example.myproject"],
    test_class = "com.example.myproject.AppTest",
    deps = [
        "AppTest.java",
        "OtherAppTest.java", # TODO: per-file tests should not have deps on other files (or maybe even themselves?)
    ],
)

java_test(
    name = "OtherAppTest",
    srcs = ["OtherAppTest.java"],
    _gazelle_imports = [
        "com.example.myproject",
        "java.lang.Exception",
        "org.hamcrest.MatcherAssert",
        "org.hamcrest.Matchers",
        "org.junit.Assert", # TODO: per-file tests should have per-test imports, excluding this one
        "org.junit.Test",
    ],
    _java_packages = ["com.example.myproject"],
    test_class = "com.example.myproject.OtherAppTest",
    deps = [
        "AppTest.java", # TODO: per-file tests should not have deps on other files (or maybe even themselves?)
        "OtherAppTest.java",
    ],
)
