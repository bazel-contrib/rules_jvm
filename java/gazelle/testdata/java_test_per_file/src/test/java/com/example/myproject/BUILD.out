load("@rules_java//java:defs.bzl", "java_test")

java_test(
    name = "AppTest",
    srcs = ["AppTest.java"],
    test_class = "com.example.myproject.AppTest",
    deps = [
        "//src/main/java/com/example/myproject",
        "@maven//:junit_junit",
        "@maven//:org_hamcrest_hamcrest_all",
    ],
)

java_test(
    name = "OtherAppTest",
    srcs = ["OtherAppTest.java"],
    test_class = "com.example.myproject.OtherAppTest",
    deps = [
        "//src/main/java/com/example/myproject",
        "@maven//:junit_junit",
        "@maven//:org_hamcrest_hamcrest_all",
    ],
)
